### PR TITLE
feat(cli): add favorite/unfavorite commands

### PR DESCRIPTION
**Pull request by johnstcn**
Part of https://github.com/coder/coder/issues/7912

Depends on https://github.com/coder/coder/pull/11791

Adds CLI commands allowing users to add/remove workspaces from their favorites.

### CODE REVIEW

File: cli/favorite.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		Handler: func(inv *clibase.Invocation) error {
+			ws, err := namedWorkspace(inv.Context(), client, inv.Args[0])
+			if err != nil {
+				return xerrors.Errorf("get workspace: %w", err)
+			}
+
+			if err := client.UnfavoriteWorkspace(inv.Context(), ws.ID); err != nil {
+				return xerrors.Errorf("unfavorite workspace: %w", err)
+			}
+			_, _ = fmt.Fprintf(inv.Stdout, "Workspace %q removed from favorites.", ws.Name)
```
Comment (johnstcn) [inline cli/favorite.go:59]:
```suggestion
			_, _ = fmt.Fprintf(inv.Stdout, "Workspace %q removed from favorites.\n", ws.Name)
```

File: cli/favorite.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		Handler: func(inv *clibase.Invocation) error {
+			ws, err := namedWorkspace(inv.Context(), client, inv.Args[0])
+			if err != nil {
+				return xerrors.Errorf("get workspace: %w", err)
+			}
+
+			if err := client.FavoriteWorkspace(inv.Context(), ws.ID); err != nil {
+				return xerrors.Errorf("favorite workspace: %w", err)
+			}
+			_, _ = fmt.Fprintf(inv.Stdout, "Workspace %q added to favorites.", ws.Name)
```
Comment (johnstcn) [inline cli/favorite.go:32]:
```suggestion
			_, _ = fmt.Fprintf(inv.Stdout, "Workspace %q added to favorites.\n", ws.Name)
```

File: cli/favorite_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	inv.Stdout = &buf
+	err := inv.Run()
+	require.NoError(t, err)
+
+	updated := coderdtest.MustWorkspace(t, memberClient, ws.Workspace.ID)
+	require.True(t, updated.Favorite)
+
+	buf.Reset()
+
+	inv, root = clitest.New(t, "unfavorite", ws.Workspace.Name)
```
Comment (mtojek) [inline cli/favorite_test.go:38]: What will happen if a user marks their workspace as "favorite" twice? Is it relevant?

**Review by mtojek — APPROVED** [review]

File: cli/favorite_test.go (hunk 3 above)
Comment (johnstcn) [reply cli/favorite_test.go:38]: Nothing interesting really will happen. It'll get updated again, which is essentially a no-op. We could check if the workspace is already favorited, but the overhead probably isn't worth the effort.

**Pull request merged**